webhookValidator.go

```diff
@@ -0,0 +1,189 @@
+// and returns as soon as the URL is considered invalid. It returns nil if the URL is
+// valid.
+func GoodURL(v []ValidURLFunc) ValidFunc {
+	return func(w Webhook) error {
+		//check Config.URL
+		parsedURL, err := url.ParseRequestURI(w.Config.URL)
+		if err != nil {
+			return fmt.Errorf("%w: %v", errInvalidURL, err)
+		}
+		for _, f := range v {
+			if f == nil {
+				continue
+			}
+			err = f(parsedURL)
+			if err != nil {
+				return fmt.Errorf("%w: %v", errInvalidURL, err)
+			}
+		}
+		//check FailureURL if it exists
+		if w.FailureURL != "" {
+			parsedFailureURL, err := url.ParseRequestURI(w.FailureURL)
+			if err != nil {
+				return fmt.Errorf("%w: %v", errInvalidFailureURL, err)
+			}
+			for _, f := range v {
+				if f == nil {
+					continue
+				}
+				err = f(parsedFailureURL)
+				if err != nil {
+					return fmt.Errorf("%w: %v", errInvalidFailureURL, err)
+				}
+			}
+		}
+		//check AlternativeURLs if they exist
+		if len(w.Config.AlternativeURLs) > 0 {
+			for _, u := range w.Config.AlternativeURLs {
+				if u != "" {
+					parsedAlternativeURL, err := url.ParseRequestURI(u)
+					if err != nil {
+						return fmt.Errorf("%w: %v", errInvalidAlternativeURL, err)
+					}
+					for _, f := range v {
+						if f == nil {
+							continue
+						}
+						err = f(parsedAlternativeURL)
+						if err != nil {
+							return fmt.Errorf("%w: %v", errInvalidAlternativeURL, err)
+						}
+					}
+				}
+			}
+		}
+		return nil
+	}
+}
+
+// Validate runs the function and returns the result.
+func (vf ValidFunc) Validate(w Webhook) error {
+	return vf(w)
+}
+
+// HTTPSOnlyEndpoints creates a Validator that considers a webhook valid if the scheme
+// of the address is https.
+func HTTPSOnlyEndpoints() ValidURLFunc {
+	return func(u *url.URL) error {
+		if u.Scheme != "https" {
+			return errURLIsNotHTTPS
+		}
+		return nil
+	}
+}
+
+// RejectHosts creates a Validator that checks the webhook address and ensures the
+// host does not contain any strings in the list of invalid hosts. It returns an error
+// if the host does include an invalid host name.
+func RejectHosts(invalidHosts []string) ValidURLFunc {
+	return func(u *url.URL) error {
+		host := u.Host
+		for _, v := range invalidHosts {
+			if strings.Contains(host, v) {
+				return errInvalidHost
+			}
+		}
+		return nil
+	}
+}
+
+// RejectALLIPs creates a Validator that checks if the given URL is an IP and returns an error
+// if it is.
+func RejectAllIPs() ValidURLFunc {
+	return func(u *url.URL) error {
+		host, _ := RemovePort(u.Host)
+		ip := net.ParseIP(host)
+		if ip != nil {
+			return errIPGivenAsHost
+		}
+		return nil
+	}
+}
+
+// RejectLoopback creates a Validator that and returns an error if the given URL is
+// a loopback address.
+func RejectLoopback() ValidURLFunc {
+	return func(u *url.URL) error {
+		host, err := RemovePort(u.Host)
+		if err != nil {
+			return err
+		} else if host == "localhost" {
+			return errLocalHostGivenAsHost
```

Comment:
```suggestion
			return errLocalhostGivenAsHost
```
Since `localhost` is one word, we don't need to capitalize `host`.